tests/resources/test_trial_metadata.py

```diff
@@ -461,3 +462,61 @@ def test_get_trial_metadata_summaries(cidc_api, clean_db, monkeypatch):
         res = client.get("/trial_metadata/summaries")
         assert res.status_code == 200
         assert res.json == result
+
+
+def test_add_new_manifest_from_json(cidc_api, clean_db, monkeypatch):
+    """Check that /trial_metadata/new_manifest endpoint behaves as expected"""
+    user_id = setup_user(cidc_api, monkeypatch)
+    make_role(user_id, CIDCRole.ADMIN.value, cidc_api)
+
+    metadata_json = {
+        "protocol_identifier": "test_trial",
+        "participants": [],
+        "shipments": [],
+        "allowed_cohort_names": ["Arm_A", "Arm_Z"],
+        "allowed_collection_event_names": ["Baseline", "Pre_Day_1_Cycle_2",],
+    }
+
+    ordered_records = OrderedDict()
+    ordered_records[ClinicalTrial] = [ClinicalTrial(protocol_identifier="test_trial")]
+    ordered_records[CollectionEvent] = [
+        CollectionEvent(trial_id="test_trial", event_name="Baseline"),
+        CollectionEvent(trial_id="test_trial", event_name="Pre_Day_1_Cycle_2"),
+        CollectionEvent(trial_id="test_trial", event_name="On_Treatment"),
+    ]
+    ordered_records[Cohort] = [
+        Cohort(trial_id="test_trial", cohort_name="Arm_A"),
+        Cohort(trial_id="test_trial", cohort_name="Arm_Z"),
+    ]
+    with cidc_api.app_context():
+        TrialMetadata(trial_id="test_trial", metadata_json=metadata_json).insert()
+        errs = insert_record_batch(ordered_records)
+        assert len(errs) == 0
+
+    client = cidc_api.test_client()
+
+    for manifest in manifests:
+        if manifest.get("status") not in [None, "qc_complete"]:
+            continue
+
+        res = client.post("/trial_metadata/new_manifest", json=manifest)
+        assert res.status_code == 200, res.json["error"]
+
+        with cidc_api.app_context():
+            md_json = TrialMetadata.select_for_update_by_trial_id(
+                "test_trial"
+            ).metadata_json
+            validate_json_blob(md_json)
+            validate_relational("test_trial")
+
+    # reusing one returns an error
+    manifest = [m for m in manifests if m.get("status") in [None, "qc_complete"]][0]
+    res = client.post("/trial_metadata/new_manifest", json=manifest)
+    assert res.status_code != 200
+    assert "already exists for trial" in res.json["error"]
+
+    # if not qc_complete, returns an error
```

Comment:
I would add a sub-test (or a new test) around failures in the payload. Think about maybe creating a manifest where the relational constraints are violated somehow, or values are malformed (bad cimac id).

Reply:
so relational constraints are violated in the test cases where the cohort or collection_event name are in the DB, via `tests/models/templates/test_csms_api.py`. malformed CIMAC IDs aren't explicitly tested, but are checked [here](https://github.com/CIMAC-CIDC/cidc-api-gae/pull/576/files#diff-f419100f9b0f7250adca8e1161d94fab18150de16ddf08fdec0df52053a6344fR228)